Containers

dfr opened the pull request.

arrowd commented:
Why some of these ports have `PKGNAMESUFFIX=	-devel`?

dfr commented:
> Why some of these ports have `PKGNAMESUFFIX= -devel`?

Mostly because they are being built from main rather than from an actual release. This is intended to match upstream's convention for unreleased code, e.g.:
```
# podman info
...
version:
  APIVersion: 4.3.0-dev
  Built: 1666624550
  BuiltTime: Mon Oct 24 16:15:50 2022
  GitCommit: 06e5870cad6f3ef2236839c7f84bbca48fbfe21c-dirty
  GoVersion: go1.19
  Os: freebsd
  OsArch: freebsd/amd64
  Version: 4.3.0-dev
```

arrowd commented:
Is this going to change in the future? If we are going to drop `PKGNAMESUFFIX` at some point, then this `-devel` suffix should be somehow a part of a version.

dfr commented:
I am expecting the next releases of both podman and buildah to build and work on FreeBSD. In theory, buildah-1.27.0 and later could be used but the main branch is more complete. For conmon, I don't know the release schedule but the selected hash is v2.1.4 plus a single commit to fix a build break so we perhaps could use that version.

arrowd commented:
I'd avoid `PKGNAMESUFFIX` in this case.

dfr commented:
What do you suggest? Perhaps for podman, 4.3.0.d to come close to upstream's convention?

arrowd commented:
Yep, or maybe even `4.3.0.dev`.

dfr commented:
If I use 4.3.0.dev, the ports build trims it to 4.3.0.d but I guess thats ok.

arrowd commented:
Huh? It seems to work for me fine.

dfr commented:
I changed the versions along these lines and removed the PKGNAMESUFFIX. I left conmon's version as 2.1.4 since its so close to the actual released version

dfr commented:
I get something like this:
```
$ cd sysutils/buildah
$ grep DISTVERSION Makefile
DISTVERSION=	1.28.1-dev
$ make package
===>  License GPLv2 accepted by the user
===>   buildah-1.28.1.d depends on file: /usr/local/sbin/pkg - found
===> Fetching all distfiles required by buildah-1.28.1.d for building
===>  Extracting for buildah-1.28.1.d
...
===>  Building package for buildah-1.28.1.d
```
The distfile is called 1.28.1-dev though:
```
$ cat distinfo
TIMESTAMP = 1666788879
SHA256 (dfr-buildah-1.28.1-dev-98b57f759_GH0.tar.gz) = 5acbc5eb4b6d3055b62fe66664b5aa17f154bc874de906a93f1dcd9735f85274
SIZE (dfr-buildah-1.28.1-dev-98b57f759_GH0.tar.gz) = 14738984
```

arrowd commented:
Use `PORTVERSION` instead of `DISTVERSION` in this case.

dfr commented:
> Use `PORTVERSION` instead of `DISTVERSION` in this case.

That worked, thanks!

arrowd commented:
My previous comments:

https://github.com/dfr/freebsd-ports/commit/e231aecd1aa46453b45ef14767d2f9242af2072a
https://github.com/dfr/freebsd-ports/commit/5d3bab3d2c403b278f46ca96e95eb99b447f4f02
https://github.com/dfr/freebsd-ports/commit/a9f05f59ea7a23efc2cea425dcf7f66713298732
https://github.com/dfr/freebsd-ports/commit/917da637b335f16f011c4906515d12f57cb3bdc4

arrowd commented on net/containernetworking-plugins/Makefile:
```diff
@@ -0,0 +1,28 @@
+GH_TAGNAME=	freebsd
+
+do-build:
+	cd ${WRKSRC} && env GO=${GO_CMD} ./build_freebsd.sh
```
Use `${SETENV}` instead of hardcoding `env`.

arrowd commented on net/containernetworking-plugins/Makefile:
```diff
@@ -0,0 +1,28 @@
+	cd ${WRKSRC} && env GO=${GO_CMD} ./build_freebsd.sh
+
+do-install:
+	${MKDIR} ${STAGEDIR}${PREFIX}/libexec/cni
```
Don't `${INSTALL_*}` commands create these dirs if they don't exist?

dfr replied:
No they don't - I get errors like this:
```
cd /wrkdirs/usr/ports/net/containernetworking-plugins/work/plugins-freebsd && install  -s -m 555 -m 755 bin/* /wrkdirs/usr/ports/net/containernetworking-plugins/work/stage/usr/local/libexec/cni
install: target directory `/wrkdirs/usr/ports/net/containernetworking-plugins/work/stage/usr/local/libexec/cni' does not exist
```
Not sure what the right way is for ports - I would prefer to use 'install -d' instead of 'mkdir -p' but I don't see a matching INSTALL_* setting in bsd.port.mk

arrowd replied:
Grepping through various ports in the tree reveals that `${MKDIR}` + `${INSTALL_PROGRAM}` is indeed way to go.

arrowd commented on sysutils/buildah/Makefile:
```diff
@@ -0,0 +1,28 @@
+		ocijail:sysutils/ocijail
+BUILD_DEPENDS=	go-md2man:textproc/go-md2man \
+		bash:shells/bash \
+		go:lang/go \
```
Replace this with `USES=go:no_targets`? This will make the port use whatever Go version is default in our Ports tree.

arrowd commented on sysutils/conmon/Makefile:
```diff
@@ -0,0 +1,19 @@
+LICENSE=	GPLv2
+
+LIB_DEPENDS=	libglib-2.0.so:devel/glib20
+BUILD_DEPENDS=	bash:shells/bash go:lang/go
```
`USES=go:no_targets`?

arrowd commented on sysutils/containers-common/Makefile:
```diff
@@ -0,0 +1,51 @@
+STORAGEVERSION=	1.43.0
+SKOPEOVERSION=	1.10.0
+
+USE_GITHUB=	nodefault
```
Move this block right after `USES`.

arrowd commented on sysutils/podman/Makefile:
```diff
@@ -0,0 +1,31 @@
+		ocijail:sysutils/ocijail
+BUILD_DEPENDS=	go-md2man:textproc/go-md2man \
+		bash:shells/bash \
+		go:lang/go \
```
`USES=go:no_targets`?

arrowd commented on sysutils/podman/Makefile:
```diff
@@ -0,0 +1,31 @@
+LICENSE=	GPLv2
+
+LIB_DEPENDS=	libgpgme.so:security/gpgme
+RUN_DEPENDS=	${LOCALBASE}/etc/containers/containers.conf.sample:sysutils/containers-common \
```
This should probably also be changed to `containers-common>=0:sysutils/containers-common`.

arrowd commented on sysutils/podman/Makefile:
```diff
@@ -0,0 +1,31 @@
+		go:lang/go \
+		gsed:textproc/gsed
+
+USES=		gmake pkgconfig python:3.7+,build shebangfix
```
Newline after this line.

arrowd commented on sysutils/podman/Makefile:
```diff
@@ -0,0 +1,31 @@
+
+USES=		gmake pkgconfig python:3.7+,build shebangfix
+SHEBANG_LANG=	python
+SHEBANG_FILES=	${WRKSRC}/hack/markdown-preprocess
```
Newline after this line.

arrowd commented on sysutils/podman/Makefile:
```diff
@@ -0,0 +1,31 @@
+USE_GITHUB=	yes
+GH_ACCOUNT=	containers
+GH_PROJECT=	podman
+GH_TAGNAME=	a77ac5be8
```
Newline after this line.

arrowd reviewed (changes requested):
Last portion of nitpicks. Thanks for working on this!

arrowd commented on net/containernetworking-plugins/Makefile:
```diff
@@ -0,0 +1,28 @@
+do-install:
+	${MKDIR} ${STAGEDIR}${PREFIX}/libexec/cni
+	${MKDIR} ${STAGEDIR}${PREFIX}/etc/containers
+	cd ${WRKSRC} && ${INSTALL_PROGRAM} -m 755 bin/* ${STAGEDIR}${PREFIX}/libexec/cni
```
`INSTALL_PROGRAM` is `-m 555`, why override  with `755` here?

arrowd commented on net/containernetworking-plugins/Makefile:
```diff
@@ -0,0 +1,28 @@
+	${MKDIR} ${STAGEDIR}${PREFIX}/libexec/cni
+	${MKDIR} ${STAGEDIR}${PREFIX}/etc/containers
+	cd ${WRKSRC} && ${INSTALL_PROGRAM} -m 755 bin/* ${STAGEDIR}${PREFIX}/libexec/cni
+	${INSTALL_DATA} -m 644 files/pf.conf.sample ${STAGEDIR}${PREFIX}/etc/containers
```
`-m 644` is already a part of `INSTALL_DATA`, so no need to have it here.

arrowd reviewed (approved):
Sort `USES` items alphabetically and this is looking good!

dfr commented:
Rebased, added 'Reviewed by'.

dfr commented:
Sorted USES

dfr commented:
Rebased, updated podman to reflect the fact that the upstream version number has changed to 4.4.0-dev (also pulling in a fix for the arm64 build). While testing, I noticed that parallel builds for podman are flaky so I added MAKE_JOBS_UNSAFE=yes there.

dwagin commented:
https://github.com/dfr/freebsd-ports/commit/a4d4118330b4ac49247ac41cd40466a4b9d4033a#r88961045

Maybe typo?

s/202210029/20221029/

arrowd commented on sysutils/podman-suite/Makefile:
```diff
@@ -0,0 +1,21 @@
+
+USES=		metaport
+
+RUN_DEPENDS=	buildah>=0.0.0:sysutils/buildah  \
```
Just `>=0` is enough.

dfr commented:
Updated podman-suite based on feedback

dwagin commented:
There are also such utilities as portclippy and portfmt, I highly recommend it.

> portfmt -i Makefile
> portclippy Makefile

dfr commented:
This is up for final review here: https://reviews.freebsd.org/D37324 - PTAL

dfr commented:
Merged by @dch
